### ISSUE TITLE
Interpolation from point data sets

### ISSUE THREAD

**Issue by RGilmore-Creare** [creare-com/podpac#34]
**Description**
When executing a node that has lat_lon coordinates as a source, the output is all 'nans'

**Describe the solution you'd like**
Get the actual data as output

**Sample Code**
```
>>> import podpac
>>> import podpac.core.data.type
>>> import numpy as np
>>> n = podpac.core.data.type.NumpyArray(source=np.arange(10), native_coordinates=podpac.Coordinate(lat_lon=((0., 1.), (1., 2.), 10)))
>>> n.execute(n.native_coordinates)
<xarray.UnitsDataArray (lat_lon: 10)>
array([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan])
Coordinates:
  * lat_lon  (lat_lon) [('lat', '<f8'), ('lon', '<f8')] (0., 1.) ...
Attributes:
    layer_style:  <podpac.core.node.Style object at 0x000000001C05A128>
    params:       None
```

**Comment by jmilloy** [comment]
See #28. I encountered this, too, and am working on this today. See also #29, which is different but related.

**Comment by mpu-creare** [comment]
I had a look at this. The problem basically seems like the Coordinate refactor didn't make it into the interpolation function. It's tests for 'lat' and 'lon' instead of 'lat_lon' and 'lon_lat' in https://github.com/creare-com/podpac/blob/develop/podpac/core/data/data.py#L510

**Comment by jmilloy** [comment]
This is fixed in ec572a2.

It's a partial fix; it won't yet work if you requesting 'lon_lat' coords from a DataSource with 'lat_lon' coords, or vice versa, but if they match, it will work.
